nameko/click_cli/myclick/paramtypes.py

```diff
@@ -0,0 +1,81 @@
+"""Custom click parameter types used in nameko.
+"""
+import sys
+
+import click
+import yaml
+
+from ..utils import import_services
+
+
+class KeyValParamType(click.ParamType):
+    """KEY=value formatted string. If only KEY is present, it gets value True.
+    """
+
+    name = "key_val"
+
+    def convert(self, value, param, ctx):
+        if "=" in value:
+            key, val = value.split("=", 1)
+            val = yaml.safe_load(val)
+            return key.strip(), val
+        else:
+            return value.strip(), True
+
+
+KEY_VAL = KeyValParamType()
+
+
+class HostPortParamType(click.ParamType):
+    """[host]:port with default 'localhost' host. Port must be int.
+    """
+
+    name = "host_port"
+
+    def convert(self, value, param, ctx):
+        host = "localhost"
+        port_str = ""
+        if ":" in value:
+            host, port_str = value.rsplit(":", 1)
+        else:
+            port_str = value
+        try:
+            port = int(port_str)
+            return host, port
+        except ValueError:
+            self.fail("%s is not a valid port number" % port_str, param, ctx)
+
+
+HOST_PORT = HostPortParamType()
+
+
+class NamekoModuleServicesParamType(click.ParamType):
+    """Nameko service(s) specified in nameko compliant module.
+    """
+
+    name = "nameko_module_services"
+
+    def __init__(self, extra_sys_paths=None):
+        """extra_sys_paths: list of paths, which are added to sys.path
+        before iporting services.
+
+        Note: syspath stays modified (side effect)
```

Comment:
To prevent this side effect, I can modify the sys.path only for the time `import_service` is called and restore it then.

Anyway, what if the code in the imported module attempts to import another module (doing that within running code, not in top level import section). If the lately imported module stays within sys.path, the code can fail.

I think, if we import the modules using modified sys.path, we shall keep the sys.path modified.

What do you think?

Reply:
Ah, quite right. It's correct that the path remains modified to allow imports at run-time.

I would remove the `(side effect)` from the comment here, because it implies it is unintentional.  It's more correct to say that we are intentionally modifying the path until the process terminates.

Reply:
done.